# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../dietlibc/dirent-align.patch
# Copyright (C) 2007 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

Fix dirent alignment as this causes unalign access warnings on mid-aged
sparc64 kernels, and strict bus errors on newer sparc64 kernels.

  - Rene Rebe <rene@exactcode.de>

--- ./dietdirent.h
+++ ./dietdirent.h
@@ -1,8 +1,8 @@
 #include <sys/shm.h>
 
 struct __dirstream {
-  int fd;
   char buf[PAGE_SIZE-(sizeof (int)*3)];
+  int fd;
   unsigned int num;
   unsigned int cur;
 };				/* stream data from opendir() */
